test/framework/bootstrap/kind_util.go

```diff
@@ -134,6 +134,7 @@ func LoadImagesToKindCluster(ctx context.Context, input LoadImagesToKindClusterI
 }
 
 // LoadImage will put a local image onto the kind node.
```

Comment:
Okay so just a recap from a user perspective:

Previously:
* loadImage:
    * if image exists locally => loads the image
    * if it doesn't => fail

With this PR:
* loadImage: 
    * if image exists locally => loads the image
    * if it doesn't => pull image
        * if it works => load the image
        * if it doesn't fail

The outcomes from a user perspective for mustLoad/tryLoad stay the same except that if the image doesn't exist we try to pull it.

I think the only issue with this could be that now you get automatically an image pulled from a registry, while before you had to explicitly make sure that the image exists in the local registry and otherwise you got a warning/error (depending on `loadBehavior`).

Impact could be that if you forget to build an image locally / or if the build fails unnoticed you'll just get a potentially entirely different image from the registry. Today you could enforce that the image must exist locally via mustLoad.

To be honest, I'm not absolutely sure if it's an issue or not, but it changes the current behavior and can lead to unexpected outcomes.

An alternative could be to introduce this behavior as an additional loadBehavior so folks can explicitly opt-in (e.g. `mustLoadPullIfMissing` / `tryLoadPullIfMissing` not the best names, but maybe we can come up with better ones).


Would like to hear @fabriziopandini's opinion on this

Reply:
Just wanted to mention that `LoadImagesToKindCluster` is used during bootstrap cluster setup and in the middle of clusterctl upgrade and self-hosted tests. If we pull in different images in the middle of those tests that is probably very hard to root cause.

Reply:
I think your recap is great and entirely correct as far as I understand it. Let's see what others think. Happy to add new config (e.g., mustLoadPullIfMissing / tryLoadPullIfMissing) with appropriate defaults to ensure behavior doesn't change for existing users.

Reply:
I think this is a nice improvement because It makes an additional try in case the image does not exist locally.

WRT to the risk of pulling an entirely different image from the registry, I'm not sure this is a problem, because we usually assign non-existing tags to the image we build, like e.g. dev. I would consider using an existing tag for something else an error/wrong practice that should be fixed vs something we to worry about.

Reply:
But I personally have a fairly good understanding how all this works and how to debug it.
If it's fine for everyone else it's fine for me as well. 

But I would really like at least to have a log statement in the case where we will pull the image instead of using the local one. This way the logs show what has been done.